Thanks for the PR here!

Was able to repro this, example below:

Modify only `versioned_feature_list.yaml` from an "all green" state with 1 versioned feature gate of k8s & hack/verify-featuregates.sh (based off of #126878):
```
aprindle@aprindle-ssd ~/kubernetes  [repro-126759]git diff
diff --git a/test/featuregates_linter/test_data/versioned_feature_list.yaml b/test/featuregates_linter/test_data/versioned_feature_list.yaml
index 58468345aac..706e6e0ba45 100644
--- a/test/featuregates_linter/test_data/versioned_feature_list.yaml
+++ b/test/featuregates_linter/test_data/versioned_feature_list.yaml
@@ -4,3 +4,8 @@
     lockToDefault: false
     preRelease: Alpha
     version: "1.30"
+  - default: true
+    lockToDefault: false
+    preRelease: Beta
+    version: "1.31"
+
```

Run `hack/verify-featuregates.sh`
```
aprindle@aprindle-ssd ~/kubernetes  [repro-126759]hack/verify-featuregates.sh 
...
found 1 features in FeatureSpecMap of func featureGates in file: /usr/local/google/home/aprindle/kubernetes/staging/src/k8s.io/component-base/metrics/features/kube_features.go
found 1 features in FeatureSpecMap var cloudPublicFeatureGates in file: /usr/local/google/home/aprindle/kubernetes/staging/src/k8s.io/controller-manager/pkg/features/kube_features.go
found 1 features in FeatureSpecMap var defaultVersionedKubernetesFeatureGates in file: /usr/local/google/home/aprindle/kubernetes/pkg/features/versioned_kube_features.go
panic: detected diff in unversioned feature list, diff: 
	  []cmd.featureInfo{
	  	{
	  		Name:     "RelaxedEnvironmentVariableValidation",
	  		FullName: "",
	  		VersionedSpecs: []cmd.featureSpec{
	  			{PreRelease: "Alpha", Version: "1.30"},
	+ 			{Default: true, PreRelease: "Beta", Version: "1.31"},
	  		},
	  	},
	  }
...

```

The error as noted in the PR states `detected diff in unversioned feature list` when the diff is in the versioned feature list which was the only file touched and relevant (the above feature is from #126878)

--- a/test/featuregates_linter/cmd/feature_gates.go
+++ b/test/featuregates_linter/cmd/feature_gates.go
@@ -167,7 +167,11 @@ func verifyOrUpdateFeatureList(rootPath, featureListFile string, update, version
 	}
 
 	if diff := cmp.Diff(featureList, baseFeatureList); diff != "" {
-		return fmt.Errorf("detected diff in unversioned feature list, diff: \n%s", diff)
+		if versioned {
+			return fmt.Errorf("detected diff in versioned feature list (%s), diff: \n%s", versionedFeatureListFile, diff)
+		} else {
+			return fmt.Errorf("detected diff in unversioned feature list (%s), diff: \n%s", unversionedFeatureListFile, diff)
+		}
 	}
 	return nil
 }

